zarr/creation.py

```diff
@@ -19,32 +22,35 @@
     normalize_storage_path,
     normalize_store_arg,
 )
+from zarr._storage.store import StorageTransformer
+from zarr.sync import Synchronizer
+from zarr.types import ZARR_VERSION, DIMENSION_SEPARATOR, MEMORY_ORDER
 from zarr.util import normalize_dimension_separator
 
 
 def create(
-    shape,
-    chunks=True,
-    dtype=None,
+    shape: Union[int, Tuple[int, ...]],
+    chunks: bool = True,
+    dtype: Optional[Union[str, npt.DTypeLike]] = None,
     compressor="default",
     fill_value: Optional[int] = 0,
-    order="C",
-    store=None,
-    synchronizer=None,
-    overwrite=False,
-    path=None,
-    chunk_store=None,
-    filters=None,
-    cache_metadata=True,
-    cache_attrs=True,
-    read_only=False,
-    object_codec=None,
-    dimension_separator=None,
-    write_empty_chunks=True,
+    order: MEMORY_ORDER = "C",
+    store: Optional[Union[str, MutableMapping]] = None,
+    synchronizer: Optional[Synchronizer] = None,
+    overwrite: bool = False,
+    path: Optional[str] = None,
+    chunk_store: Optional[MutableMapping] = None,
```

Comment:
maybe there's an opportunity to specialize the `MutableMapping` annotations with the type parameters for the keys and values? e.g., `MutableMapping[str, bytes]`

Reply:
I might leave that for another PR to keep this one simple?

Reply:
that's fine!